OpenBB-finance/OpenBBTerminal
# Update latest allowed python version to 3.10.x

# Description

This PR raises the latest allowed python version 3.10.x which is the latest minor version.

# How has this been tested?

I have run the test suite using `pytest tests` and although tests were skipped, there were no failures. This was done on macOS 12.4 on which python is managed using asdf and the python plugin for asdf.

To reproduce

1. Install a current 3.10.x version of python, in my case I use the [asdf version manager](https://asdf-vm.com/) and the associated python [plugin](https://asdf-vm.com/manage/plugins.html).
1. Confirm you are using that python version in your current shell using `python --version`
1. [Install poetry](https://python-poetry.org/docs/#installation)
1. Install all dependencies including optional ones

   ```shell
   poetry install --no-root -E prediction -E bots -E jupyterlab
   ``` 

1. Activate the poetry shell

   ```shell
   poetry shell
   ```

1. Run test suite

   ```shell
   pytest tests
   ```

This should enable you to confirm (hopefully) that all dependencies install cleanly using just poetry and also that the test suite does not have any failures, although some tests are skipped.


# Others
- [x] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [x] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.
